tools/generate-test-case

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python3
```

Comment:
I think you did a `git add` rather than `git mv`, which makes both the patches a bit hard to follow. If you first do the rename, then do a second patch to add a boot type to this script (or the other way around), and then add the new script in the final patch, I think it will be clearer.
Very happy about the actual change here though. Love the in-line documentation.

Reply:
Improved the test case